testplan/testing/multitest/base.py

```diff
@@ -909,7 +909,18 @@ def _run_suite_related(self, testsuite, method):
 
         with method_report.timer.record("run"):
             with method_report.logged_exceptions():
-                testsuite_method(*method_args)
+                time_restriction = getattr(testsuite_method, "timeout", None)
+                if time_restriction:
+                    # pylint: disable=unbalanced-tuple-unpacking
+                    executed, execution_result = timing.timeout(
```

Comment:
we cannot use it directly, because the method signature check will fail for decorated methods. also, `time.timing` is used for methods that already bound, while in testsuite definition those are unbound methods.